### PR TITLE
initial working copy

### PR DESCRIPTION
**Pull request by adamleff**
Lacking tests, but successfully creating machines, snapshotting them, and uploading images to GCP Storage.

### CODE REVIEW

File: libraries/_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      run_list = []
+      run_list << 'marketplace_gce::bootstrap'
+      run_list << 'marketplace_image::_publisher'
+      run_list << 'marketplace_gce::create_snapshot' if snapshot
+      run_list << security_recipe if security
+      run_list << audit_recipe if audit
+      run_list
+    end
+
+    def verify_terraform_installed!
```
Comment (ryancragun) [inline libraries/_helpers.rb]: :lollipop:

File: libraries/_helpers.rb [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        Chef::Log.info(shell_out("terraform plan -state #{state_file} #{config_dir}"))
+
+        Chef::Log.info("Running 'terraform apply' for #{opts[:type]}...")
+        Chef::Log.info(shell_out("terraform apply -state #{state_file} #{config_dir}"))
+      end
+    end
+
+    def shell_out(cmd)
+      shell = Mixlib::ShellOut.new(cmd, timeout: 1800)
+      shell.live_stream = STDOUT
```
Comment (ryancragun) [inline libraries/_helpers.rb]: Not actually an issue, but generally it's better to use `$stdout` than to directly refer to `STDOUT`

File: libraries/_helpers.rb [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+          'google' => {
+            'credentials' => %Q<${file("#{gce_credentials_file}")}>,
+            'project'     => project,
+            'region'      => region
+          }
+        }
+      }
+    end
+
+    def terraform_instance_config
```
Comment (ryancragun) [inline libraries/_helpers.rb]: :dog2: such hash

**Comment by ryancragun** [comment]
:rocket: The only question I have is the resource name.  What does one call a GCE image?  Google Compute Images?  Perhaps it should be named `markeplace_gci` to better emphasize that this resource is for creating images?

File: libraries/_helpers.rb (hunk 1 above)
Comment (adamleff) [inline libraries/_helpers.rb]: Yeah, I'll add something there. :)

File: libraries/_helpers.rb (hunk 2 above)
Comment (adamleff) [inline libraries/_helpers.rb]: Heh, I just copied what mixlib-shellout does in its own code.

File: Berksfile [hunk 4]
```diff
@@ -0,0 +1,5 @@
+source 'https://supermarket.chef.io'
+
+metadata
+
+cookbook 'marketplace_image', github: 'chef-partners/marketplace_image'
```
Comment (ryancragun) [inline Berksfile]: We don't need this anymore, right?

File: metadata.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+name 'marketplace_gce'
+maintainer 'Chef Partner Engineering'
+maintainer_email 'partnereng@chef.io'
+license 'Apache 2.0'
+description 'Installs/Configures marketplace_gce'
+long_description 'Installs/Configures marketplace_gce'
+version '0.1.0'
+
+depends 'marketplace_image'
```
Comment (ryancragun) [inline metadata.rb]: Ditto

**Comment by ryancragun** [comment]
:ship:

**Pull request merged**